src/js/libgeo/GeoOps.js

```diff
@@ -625,6 +625,144 @@ geoOps.PointOnSegment.updatePosition = function(el) {
+    }
+    return [m,n];
+};
+
+geoOps._helper.projectPointToArc = function(arc,P){
+    var conic = arc.matrix;
+    var cen = geoOps._helper.CenterOfConic(conic);
+
+    var diam = List.cross(cen, P);
+ //   Render2D.drawline(diam);
+//    debugger;
+
+    var isec = geoOps._helper.IntersectLC(diam, arc.matrix);
+
+
+    var crA = geoOps._helper.PointOnArcCr(arc, isec[0]);
+    var prodA = CSNumber.mult(crA[0], crA[1]);
+   // console.log("pA",niceprint(prodA));
+    if (crA[0].value.real < 0) prodA = CSNumber.neg(prodA);
+
+    var crB = geoOps._helper.PointOnArcCr(arc,isec[1]);
+    var prodB = CSNumber.mult(crB[0], crB[1]);
+//    console.log("pB",niceprint(prodB));
+    if (crB[0].value.real < 0) prodB = CSNumber.neg(prodB);
+
+    
+    var erg;
+    if(prodA.value.real > 0 && prodB.value.real > 0){
+        console.log("hit1");
+        var d1 = List.projectiveDistMinScal(P, isec[0]);
+        var d2 = List.projectiveDistMinScal(P, isec[1]);
+
+        erg = d1 < d2 ? isec[0] : isec[1];
+    }
+    else if (prodA.value.real < 0 && prodB.value.real < 0){
+        console.log("hit rand proj");
+    var d1 = List.projectiveDistMinScal(pos, arc.startPoint);
+    var d2 = List.projectiveDistMinScal(pos, arc.endPoint);
+
+    erg = d1 < d2 ? arc.startPoint : arc.endPoint;
+    }
+    else if(prodA.value.real < 0){ 
+        console.log("hit3");
+        erg = isec[1];
+    }
+    else //(prodB.value.real < 0)
+    {
+        console.log("hit4");
+        erg = isec[0];
+    }
+
+
+    return List.normalizeMax(erg);
+};
+
+
+geoOps.PointOnArc = {};
+geoOps.PointOnArc.kind = "P";
+geoOps.PointOnArc.signature = ["C"];
+geoOps.PointOnArc.isMovable = true;
+geoOps.PointOnArc.initialize = function(el) {
+    var arc = csgeo.csnames[el.args[0]];
+    var p = geoOps._helper.initializePoint(el);
+    p = geoOps._helper.projectPointToArc(arc, p);
+
+    var cr = geoOps._helper.PointOnArcCr(arc, p);
+    cr = CSNumber.mult(cr[0],cr[1]);
+    putStateComplexNumber(cr);
+};
+geoOps.PointOnArc.getParamForInput = function(el, pos) {
+    var arc = csgeo.csnames[el.args[0]];
+    var cr = geoOps._helper.PointOnArcCr(arc, pos);
+    cr = CSNumber.mult(cr[0],cr[1]);
+
+    return cr;
+};
+geoOps.PointOnArc.getParamFromState = function(el) {
+    return getStateComplexNumber();
+};
+geoOps.PointOnArc.putParamToState = function(el, param) {
+    putStateComplexNumber(param);
+};
+geoOps.PointOnArc.updatePosition = function(el,isMover) {
+    var arc = csgeo.csnames[el.args[0]];
+    var A = arc.startPoint;
+    var C = arc.endPoint;
+    var B = arc.viaPoint;
+    var II = List.ii;
+
+    var conic = arc.matrix;
+
+    var cr = getStateComplexNumber();
+    putStateComplexNumber(cr); // copy parameter
+
+    var erg;
+    if( isMover){
+        var point = geoOps._helper.initializePoint(el);
+        console.log(niceprint(List.normalizeZ(point)));
+        erg = geoOps._helper.projectPointToArc(arc,point);
+        // projection should be done here
```

Comment:
What is the motivation for the case distinction here? I'd have thought that we could always compute the result based on the arc and the cross ratio.